agent/rpc_to.go

```diff
@@ -71,12 +72,54 @@ func (a *orbAgent) sendGroupMembershipReq() error {
 	if token := a.client.Publish(a.rpcToCoreTopic, 1, false, body); token.Wait() && token.Error() != nil {
 		return token.Error()
 	}
-
 	return nil
 }
 
+func (a *orbAgent) sendGroupMembershipReq() error {
+	defer a.retryGroupMembershipRequest()
+	return a.sendGroupMembershipRequest()
+}
+
+func (a *orbAgent) retryGroupMembershipRequest() {
+	if a.groupRequestTicker == nil {
+		a.groupRequestTicker = time.NewTicker(retryRequestFixedTime * retryRequestDuration)
+	}
+
+	go func() {
+		defer a.groupRequestTicker.Stop()
+		defer func(t time.Time) {
+			a.logger.Info("execution period of the re-request of retryGroupMembership", zap.Duration("waiting period", time.Now().Sub(t)))
+		}(time.Now())
+		lastT := time.Now()
+		for calls := 1; calls <= retryMaxAttempts; calls++ {
+			select {
+			case <-a.groupRequestSucceeded:
+				a.groupRequestTicker.Stop()
+				return
+			case t := <-a.groupRequestTicker.C:
+				a.logger.Info("agent did not received any group membership from fleet, re-requesting", zap.Duration("waiting period", lastT.Sub(t)))
+				duration := retryRequestFixedTime + (calls * retryDurationIncrPerAttempts)
+				a.groupRequestTicker.Reset(time.Duration(duration) * retryRequestDuration)
+				err := a.sendGroupMembershipRequest()
+				if err != nil {
+					a.logger.Error("failed to send group membership request", zap.Error(err))
+					return
+				}
+				lastT = t
+			}
+		}
+		a.logger.Warn(fmt.Sprintf("retryGroupMembership retried %d times and still got no response from fleet", retryMaxAttempts))
```

Comment:
what actually happens in this case afterwards? should we stop the agent?

Reply:
I don't think we should stop the agent on this, maybe another reset, can fix, or if there is another issue with mqtt connection this can actually stop the container on other functions, the re-request should not have this responsability